Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -498,16 +498,12 @@ bool does_self_intersect(const FaceRange& face_range,
 {
   CGAL_precondition(CGAL::is_triangle_mesh(tmesh));
 
-  try
-  {
+  try {
     CGAL::Emptyset_iterator unused_out;
     internal::self_intersections_impl<ConcurrencyTag>(face_range, tmesh, unused_out, true /*throw*/, np);
-  }
-  catch(CGAL::internal::Throw_at_output_exception&)
-  {
+  } catch(const std::exception&) {
     return true;
   }
-
   return false;
 }
```

Comment:
That looks wrong to me. Now the function `does_self_intersect` will return `true` as soon as any exception deriving from `std::exception` is thrown. If there is a `std::bad_alloc` during the execution, then the function will catch it, and returns `true`, as if the surface self-intersected.

The function is supposed to return `true` only if an exception of the precise type `CGAL::internal::Throw_at_output_exception` is thrown, and for any other type of exception, then exception should not be caught.

Reply:
I see.

Reply:
Ok, I now start from the most specific exception and then only handle TBB, however the comparison between the `typeid name` and ` exception name` do not work for some reason, always returning `false`. I will have a look at what can be done with that.